### PR TITLE
Upgrade .NET Core to 3.1 and replace external JSON library

### PR DESCRIPTION
**Pull request by NucsuM**
This PR updates the following parts:

* upgrade from .NET Core `2.2.401` to `3.1.200` (in relation to this [guide](https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-3.1&tabs=visual-studio))
* Replace `Newtonsoft Json.NET` with  the internal JSON library (`System.Text.Json`) where it is possible. `Newtonsoft Json.NET` is still included because it is needed for pretty printing API response

### CODE REVIEW

**Comment by tosch** [comment]
Can you please add some style to the code? Blank lines between class definitions, for instance and such stuff. Is there some automagic linter for C#?

**Comment by NucsuM** [comment]
> Can you please add some style to the code? Blank lines between class definitions, for instance and such stuff. Is there some automagic linter for C#?

This is a good question, I have never heared that people use a C# Linter, but I will have a short look in the new sage c# service repositories, what they might use. 

*Addition:*
In some internal repositiories I found the usage of `StyleCop`. I tryed it on this PR and I got 200+ offences. I did not find a quick way to do some "autocorrect", but using this seems to be very helpful.

File: app/Startup.cs [hunk 1]
```diff
@@ -44,14 +70,16 @@ public void ConfigureServices(IServiceCollection services)
...
-        using (StreamReader file = File.OpenText(getPathOfConfigFile()))
-        using (JsonTextReader reader = new JsonTextReader(file))
-        {
-          JObject configObj = (JObject)JToken.ReadFrom(reader);
-          config_client_id = (string)configObj["config"]["client_id"];
-          config_client_secret = (string)configObj["config"]["client_secret"];
-          config_calback_url = (string)configObj["config"]["callback_url"];
-        }
+      {
+          //using (StreamReader file = File.OpenText(getPathOfConfigFile()))
```
Comment (cpetschnig) [inline app/Startup.cs]: Can this be removed?

**Review by cpetschnig — APPROVED** [review]

File: app/Startup.cs (hunk 1 above)
Comment (NucsuM) [reply app/Startup.cs]: sorry, I removed it.

**Comment by tosch** [comment]
This has been QAed successfully. [This is the test session](https://jira.sage.com/secure/ViewSession.jspa?testSessionId=93785).

**Pull request merged**